ui/app/src/modules/Content/Dependencies/DependenciesDialogUI.tsx

```diff
@@ -0,0 +1,473 @@
+    alignItems: 'center',
+    padding: '0 10px 0 0',
+    '& .label': {
+      fontWeight: '600',
+      marginRight: '10px',
+      color: palette.black,
+      fontSize: '16px'
+    },
+    '& .item-icon': {
+      color: palette.teal.main,
+      marginRight: '10px',
+      width: '20px',
+      height: '20px'
+    },
+    '& .item-title': {
+      marginRight: '25px'
+    }
+  },
+  selectedItemEditIcon: {
+    fontSize: '14px',
+    color: palette.gray.medium5,
+    width: '16px',
+    height: '16px'
+  },
+  dependenciesList: {
+    backgroundColor: palette.white,
+    padding: 0,
+    borderRadius: '5px 5px 0 0',
+    overflowY: 'auto'
+  },
+  dependenciesListItem: {
+    boxShadow: '0 1px 1px #EBEBF0',
+    padding: 0,
+    height: '70px'
+  },
+  dependenciesCompactListItem: {
+    height: '43px'
+  },
+  listItemPreview: {
+    width: '100px',
+    height: '70px',
+    borderRadius: 0
+  },
+  listItemContent: {
+    paddingLeft: '15px'
+  },
+  compactViewAction: {
+    marginRight: 'auto'
+  },
+  showTypesSelect: {
+    '& > .MuiRadio-root': {
+      display: 'none'
+    }
+  },
+  showTypesMenu: {
+    '& .MuiListItem-root': {
+      padding: '0 10px',
+      fontSize: '14px',
+      '& > .MuiRadio-root': {
+        padding: '6px',
+        '& .MuiSvgIcon-root': {
+          width: '16px',
+          height: '16px'
+        }
+      }
+    }
+  },
+  suspense: {
+    height: '100%',
+    flexDirection: 'unset'
+  }
+}));
+
+interface DependenciesListProps {
+  resource: Resource<Item[]>;
+  state: any;
+  setState: Function;
+  assetsTypes: any;
+  isEditableItem: Function;
+  handleEditorDisplay: Function;
+}
+
+function DependenciesList(props: DependenciesListProps) {
+  const {
+    resource,
+    state,
+    setState,
+    assetsTypes,
+    isEditableItem,
+    handleEditorDisplay
+  } = props;
+  const classes = dependenciesDialogStyles({});
+  const dependencies: Item[] = resource.read();
+
+  return (
+    <List className={classes.dependenciesList}>
+      {
+        dependencies
+          .filter(dependency => assetsTypes[state.showTypes].filter(dependency))
+          .map(dependency => {
+            return (
+              <ListItem
+                key={dependency.uri}
+                className={`${classes.dependenciesListItem} ${(state.compactView) ? classes.dependenciesCompactListItem : ''}`}
+              >
+                {
+                  isImage(dependency.uri) && !state.compactView &&
+                  <ListItemAvatar>
+                    <Avatar className={classes.listItemPreview} src={dependency.uri}/>
+                  </ListItemAvatar>
+                }
+                <ListItemText
+                  className={classes.listItemContent}
+                  primary={dependency.internalName}
+                  secondary={(!state.compactView) ? dependency.uri : null}
+                />
+
+                <ListItemSecondaryAction>
+                  <PopupState variant="popover">
```

Comment:
What's the popup here for? What's justifies using this over simply a menu?
We also have a menu-generator component. We should probably use that here.